### PR TITLE
Allow for use in other Packages

### PR DESCRIPTION
**Pull request by spprichard**
- Added library product to allow other packages to use this code

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -5,11 +5,18 @@ import PackageDescription
...
+    ],
     products: [
         .executable(
             name: "SQLiteVecCLI",
             targets: ["SQLiteVecCLI"]
         ),
+        .library(
+            name: "SQLiteVec",
+            targets: ["SQLiteVec"]
+        ),
```
Comment (jkrukowski) [inline Package.swift:16]: good call, could you please add as well

```swift
.library(
    name: "CSQLiteVec",
    targets: ["CSQLiteVec"]
),
```

File: Package.swift [hunk 2]
```diff
@@ -5,11 +5,18 @@ import PackageDescription
 
 let package = Package(
     name: "SQLiteVec",
+    platforms: [
+        .iOS(.v16)
+    ],
```
Comment (jkrukowski) [inline Package.swift]: I wouldn't add it. Specifying `platforms` (according to the [docs](https://github.com/swiftlang/swift-package-manager/blob/main/Documentation/PackageDescription.md#supportedplatform)) means

> By default, the Swift Package Manager assigns a predefined minimum deployment version for each supported platforms unless you configure supported platforms using the platforms API. This predefined deployment version is the oldest deployment target version that the installed SDK supports for a given platform. 

So it should work on iOS without explicitly specifying it here

**Review by jkrukowski — COMMENTED** [review]
Thanks for the contribution! Left some comments, lmk what you think

File: Package.swift (hunk 2 above)
Comment (spprichard) [reply Package.swift]: Ya I was curious about that. Yup not problem, I'll remove

File: Package.swift (hunk 1 above)
Comment (spprichard) [reply Package.swift:16]: Sure no problem 👍

**Pull request merged**